testing/engine/allow.go

```diff
@@ -127,12 +127,56 @@ var _ = profile.RegisterProfile(profile.Profile{
 						},
 					},
 				},
+				{
+					Stage: profile.SubStage{
+						Input: profile.StageInput{
+							URI:    "/response_allow",
+							Method: "POST",
+							Headers: map[string]string{
+								"Content-type": "application/x-www-form-urlencoded",
+							},
+							Data: "response_allow",
+						},
+						Output: profile.ExpectedOutput{
+							TriggeredRules: []int{
+								70,
+							},
+							NonTriggeredRules: []int{
+								71,
+							},
+							Interruption: &profile.ExpectedInterruption{
+								Status: 500,
+								Data:   "",
+								RuleID: 70,
+								Action: "deny",
+							},
+						},
+					},
+				},
 			},
 		},
 	},
+	// TODO(MultiPhase)[MovingIntoAllowedPhase]: see rules 45 and 46. Rule 45 allows all the request phases (1 and 2)
```

Comment:
Some ideas.

We should keep in mind that in the end, our main target is CRS so optimizing the behavior for CRS while still correctly handling rules is OK.

Bluntest hammer: disable multiphase evaluation if any rule contains an allow action. AFAICT, CRS does not contain any allow actions, which makes senes, and is why FTW is working I guess.

It's too blunt I guess, the main reason for allow would be for users to be able to override CRS behavior for their unique needs.

So if we can come up with a user experience that provides this and document it, and return errors otherwise, then it could be OK.

For example, implement the following type of rules

Multiphase is enabled if allow actions are only in phase 1
Multiphase is enabled if allow actions are only in the lowest-priority rules of a given phase (we could change our inferrence to only allow upgrading phase 2 to phase 1, phase 4 to phase 3, rather than having e.g. phase 3 upgrade to phase 1, that doesn't seem to have much use case)
...any other condition that seems to make sense

We should be able to detect cases where these don't hold and return a configuration error with guidance on how configs can be tweaked to define an allow rule that works with multiphase. 

This may not allow bypassing every single situation - I think the above would not allow a request body to contain a log4shell string, and to also contain the string IAmSafe and be allowed by a rule as log4shell would always win. An obvious solution to this doesn't come to mind - but I don't know if this is useful enough to worry, as long as we return a configuration error rather than ignoring, it should be OK. I would suspect the vast majority of allow to be in phase 1, based on headers/cookies? Don't know but basically if the cases we can't enable multiphase for are corner cases, it should be OK, we just have to make sure we tell the user why when we can't.